File: app/src/entourage/java/social/entourage/android/user/role/UserRolesFactory.java
package social.entourage.android.user.role;

import social.entourage.android.R;

/**
 * Entourage-related user roles
 * Created by Mihai Ionescu on 21/05/2018.
 */
public class UserRolesFactory extends BaseUserRolesFactory {

    // ----------------------------------
    // SINGLETON
    // ----------------------------------

    private static final UserRolesFactory ourInstance = new UserRolesFactory();

    public static UserRolesFactory getInstance() {
        return ourInstance;
    }

    private UserRolesFactory() {
        register(new UserRole("ambassador", R.string.role_ambassador, R.color.profile_role_ambassador, true));
    }

}

File: app/src/main/java/social/entourage/android/user/role/UserRolesFactory.kt
package social.entourage.android.user.role

import social.entourage.android.R
import java.util.ArrayList

/**
 * Entourage-related user roles
 * Created by Mihai Ionescu on 21/05/2018.
 */
object UserRolesFactory  {
    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------
    private val userRoles = ArrayList<UserRole>()

    // ----------------------------------
    // PUBLIC METHODS
    // ----------------------------------
    fun register(userRole: UserRole) {
        userRoles.add(userRole)
    }

    fun findByName(name: String): UserRole? {
        for (userRole in userRoles) {
            if (name.equals(userRole.name, ignoreCase = true)) {
                return userRole
            }
        }
        return null
    }

    init {
        register(UserRole("ambassador", R.string.role_ambassador, R.color.profile_role_ambassador, true))
    }
}